presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaSplitManager.java

```diff
@@ -100,10 +99,28 @@ public ConnectorPartitionResult getPartitions(ConnectorTableHandle tableHandle,
                         log.warn("No leader for partition %s/%s found!", metadata.topic(), part.partitionId());
                     }
                     else {
+                        // make sure the first broker is the partition leader
+                        ImmutableList.Builder<HostAddress> partitionNodesBuilder = ImmutableList.builder();
+                        HostAddress partitionLeader = brokerToHostAddress(leader);
+
+                        boolean isLeaderFirst = true;
+                        if (part.isr().get(0).id() != leader.id()) {
```

Comment:
I don't understand this part. What it does is to ensure that the leader is always the first element in the list. That seems to imply that the leader is special in the isr set, so the way it is used may simply be incorrect. This feels like a convoluted way to talk only to the leader.